x-pack/plugins/infra/public/containers/logs/log_analysis/log_analysis_module.tsx

```diff
@@ -123,6 +123,9 @@ export const useLogAnalysisModule = <JobType extends string>({
       createPromise: async () => {
         return await moduleDescriptor.cleanUpModule(spaceId, logViewId, services.http.fetch);
       },
+      onReject: (e) => {
+        throw new Error(`Failed to clean up previous ML job: ${e}`);
+      },
```

Comment:
This part was weird, if I don't do this, then the Promise is not rejected even if the API call throws an error (which is why we didn't see this API move).
@weltenwort Do you know if that's expected?

Reply:
My _guess_ is that `useTrackedPromise` is swallowing errors and expecting callers to look for and respond to the "rejected" state instead, https://github.com/miltonhultgren/kibana/blob/d91fe9fc92133cfa52c4b9aff664018b241f2ac3/x-pack/plugins/infra/public/utils/use_tracked_promise.ts#L222

Reply:
Not sure which makes more sense here exactly, but you could look at `cleanUpModuleRequest.state` for "rejected" and respond, rather than throwing this new error?

Reply:
I'm throwing here because this function is called further down and then as a Promise with a `catch` to update the reducer state, if I don't catch I'm outside of that flow and the setup call is done ahead of time which causes an inconsistent state.